test/pkg.jl

```diff
@@ -398,4 +398,73 @@ temp_pkg_dir() do
         @test contains(msg, "$package is not an installed package")
         @test !contains(msg, "signal (15)")
     end
+
+    # partial Pkg.update
+    begin
+        nothingtodomsg = "INFO: No packages to install, update or remove\n"
+
+        ret, out, err = @grab_outputs begin
+            Pkg.rm("Example")
+            Pkg.add("Example")
+        end
+        @test ret == nothing && out == ""
+        @test contains(err, "INFO: Installing Example v")
+
+        ret, out, err = @grab_outputs Pkg.update("Example")
+        @test ret == nothing && out == ""
+        @test contains(err, nothingtodomsg)
+
+        ret, out, err = @grab_outputs begin
+            Pkg.rm("Example")
+            Pkg.add("Example", v"0", v"0.4.1-")
```

Comment:
Accepting multiple versions here was new to me - doesn't need to be done in this PR, but might be good to spell out how the versions are interpreted when > 2 are provided. Looks from the code like they're taken as pairs of bounds for separate allowable intervals?